### ISSUE TITLE
[Feature Request] Transparent Navigation bar

### ISSUE THREAD

**Issue by shuvashish76** [TeamAmaze/AmazeFileManager#2747]
**With dark theme**
![Screenshot_2021-08-23-08-54-45](https://user-images.githubusercontent.com/48226312/130386882-2bb652eb-275e-4a39-8cbf-ba69b73aa826.jpeg)
![Screenshot_2021-08-23-08-55-28](https://user-images.githubusercontent.com/48226312/130386912-ee442189-7af3-4f9a-b3ed-370a7fc7568e.jpeg)

**With light theme**
![Screenshot_2021-08-23-08-57-04](https://user-images.githubusercontent.com/48226312/130386973-17cd91a2-2a2f-41e3-80b3-b4e0cc3566e9.jpeg)
![Screenshot_2021-08-23-08-58-00](https://user-images.githubusercontent.com/48226312/130386987-aa76ecd4-5a86-40db-832f-a737e7e652cf.jpeg)

Comparison vs [MaterialFiles](https://github.com/zhanghai/MaterialFiles)

Currently we do have _Colored Navigation bar_ option but colors are solid colors. Colored Navigation bar with option to choose from solid/transparent color would be nice. Transparent navigation bar looks modern imo.

**Comment by abdullah951** [comment]
Hi, I would like to work on it.

**Comment by TranceLove** [comment]
@abdullah951 Please do.

**Comment by ANUKULJHA1506** [comment]
@ashway171 can you assign me this.

**Comment by VishnuSanal** [comment]
> can you assign me this.

@ashway171 Please confirm if you're still working on this issue. @ANUKULJHA1506 We'll assign you after their confirmation or if there's no reply.

**Comment by ANUKULJHA1506** [comment]
@VishnuSanal ok

**Comment by ashway171** [comment]
> > can you assign me this.
> 
> @ashway171 Please confirm if you're still working on this issue. @ANUKULJHA1506 We'll assign you after their confirmation or if there's no reply.

Yeah, I'm on it... Just some errors and we are good to go

**Comment by VishnuSanal** [comment]
> Yeah, I'm on it...

Okay, so I think @ANUKULJHA1506 can look into other [good first issues](https://github.com/TeamAmaze/AmazeFileManager/labels/good%20first%20issue) :)

**Comment by malik0011** [comment]
Hey, is it still open? I would like to work on it!

**Comment by VishnuSanal** [comment]
> Hey, is it still open? I would like to work on it!

Yes, and feel free to do!

**Comment by VishnuSanal** [comment]
@malik0011 Any updates here?

**Comment by Trust04zh** [comment]
Hi, I'd like to work on this issue if nobody working on it.

**Comment by VishnuSanal** [comment]
> Hi, I'd like to work on this issue if nobody working on it. 

@Trust04zh Sure, please do.

**Comment by mohit-2003** [comment]
I don't understand what is the issue, can you please explain it?
**what i have to do ?**

**Comment by VishnuSanal** [comment]
> I don't understand what is the issue, can you please explain it? **what i have to do ?**

@Trust04zh is working on this as of now. @mohit-2003 can take this over if @Trust04zh is not working on this anymore.

**Comment by Trust04zh** [comment]
I'm currently still working on this and try figuring out how to apply transparent navigation bar with the original UI designing, I am not so familiar with Android UI designing, the process may be kinda slow.

**Comment by VishnuSanal** [comment]
Okay, so I think @mohit-2003 can look into other [good first issues](https://github.com/TeamAmaze/AmazeFileManager/labels/good%20first%20issue%F0%9F%92%AC) :)

**Comment by Trust04zh** [comment]
@shuvashish76 Thank you for your patient explanation! 

I noticed the facts just as you describe about the current behaviors (also I think the behavior when turning off may not be as expected, as the original color is not always black, this might be another issue), while I wonder if it would be better to add an independent option for setting transparent navigation bar, as the "Colored Navigation Bar" setting sounds not for setting the navigation bar to be transparent or not, I think. 

So I think there might be another checkbox below "Colored Navigation Bar" to set the navigation bar to be transparent, I would be thankful if you could share your option with me.

And What I'm currently figuring out is about the code implementation for this feature, simply set an transparent `argb` color with api `Window.setNavigationBarColor` won't set a truly transparent color, and I'm reading references and testing to let it work.

**Comment by shuvashish76** [comment]
> the behavior when turning off may not be as expected

Agreed, but for example by default Material Files show transparent navigation bar & no special settings to turn that off/on. So why not follow same behaviour for Amaze (_off/transparent by default_)?

> better to add an independent option for setting transparent navigation bar

- I like Amaze's minimal & organised settings & IMO another setting just for that is unnecessary.
- Ok what would be the navigation bar color for Amaze, if you turn off both `Colored Navigation bar` & `Transparent Navigation bar`? Black?

**Comment by Trust04zh** [comment]
Sorry for my late reply.

> Ok what would be the navigation bar color for Amaze, if you turn off both Colored Navigation bar & Transparent Navigation bar? Black?
System default color, white for light theme while black for dark theme, which are the colors you launch Amaze with corresponding system theme and with `Colored Navigation bar` disabled, I guess this might be the expected behavior than simply black.

> I like Amaze's minimal & organised settings & IMO another setting just for that is unnecessary.
I agree that another option can be unnecessary, while I'm not sure if it is ok if there's no option for "solid and system default navigation bar color".

**Comment by Trust04zh** [comment]
some report for my process: 

I implemented transparent navigation bar by modifying styles so that navigation bar color become transparent, and enabled some flags to allow views to draw under system bars, but I still need to specify what should be drawn under system bars and what should not, or it will simply be like this.

![图片](https://user-images.githubusercontent.com/36195713/161429687-eb28eaf5-38b6-432a-865f-9f94ed26cd90.png)


I've found some advises about using `windowInsets` to control the layout to solve this, also I read some source code of `MaterialFiles` and it seems that it use `windowInset` and padding to control the layout, while I didn't finish it yet.

**Comment by VishnuSanal** [comment]
@Trust04zh Are you still working on this?

**Comment by Trust04zh** [comment]
@VishnuSanal  
No, I didn't find an easy approach to implement this feature.
Two references that maybe helpful:
For transparent system bars
https://medium.com/androiddevelopers/gesture-navigation-going-edge-to-edge-812f62e4e83e
For only draw behind status bar (so can navigation bar)
https://stackoverflow.com/questions/68451704/transparent-status-bar-with-visible-navigation-bar#68492550

**Comment by PK284** [comment]
**If anybody is not assigned to this Can I solve this issue?**
